### ISSUE TITLE
Incorrect user profile image shown in notifications

### ISSUE THREAD

**Issue by CombeeMike** [signalapp/Signal-Android#10329]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I recently installed Signal on a new Pixel 4a with Android 11. When receiving new messages, all notifications show the same (incorrect) user profile picture in the notification drawer. I'm not 100% sure if this is the case for all contacts. It definitely happens for 3 of them and they all show the same incorrect picture. The picture shown was used for at least some (not sure if all) contacts in the "in app" contact list as well short after installation but that "fixed itself" after a few moments.

In short:
* Incorrect profile pictures were shown in contact list short after installation (only lastet a few moments & "fixed itself")
* Correct profile pictures are shown in contact list now
* Incorrect profile picture is still shown in notifications

It seems like there's some internal cache which was incorrectly initialized & only updated for the contact list afterwards. Can I maybe somehow force a clean/rewrite of such a cache? I already tried restarting the phone to no avail.

### Steps to reproduce
- Receive message from someone

**Actual result:** Incorrect profile picture is shown in notification drawer
**Expected result:** Correct profile picture in notification drawer

### Screenshots
![Screenshot_20201223-080833](https://user-images.githubusercontent.com/30658772/102970940-b7bb0f80-44f8-11eb-8562-b4013399114c.png)

### Device info
**Device:** `Google Pixel 4a`
**Android version:** `11`, Builr Nr. `RQ1A.201205.008`
**Signal version:** `5.0.8`

### Link to debug log
https://debuglogs.org/b78310a068eba400dc6c1e152262bd52c6bbaa1fa69ea24e02441e67dc79eb27

**Comment by alex-signal** [comment]
Hi! Could you please verify if these contacts are in your system contacts or not? And if so, are the photos you're seeing in the conversation list (the corrected ones) being pulled from your system contacts?

**Comment by Cthulku** [comment]
My apologies if this should be addressed elsewhere, but is there a way to have Signal use its own profile picture rather than the system contact one, if available? I don't get why having a Signal-exclusive profile picture is a feature if it is overridden by the system contact one.

**Comment by alex-signal** [comment]
@Cthulku Outside of modding / building your own version of the app (I think there's like, one line you'd commend out in Recipient.java?) no. Showing system contact photos vs profile photos has been discussed in the past, and we've at one point swapped how it works, but ended up changing it back after listening to community feedback. For future reference, the best place to search for stuff like this is over at community.signalusers.org :)

**Comment by Cthulku** [comment]
Ah, thank you!

**Comment by CombeeMike** [comment]
@alex-signal Hi! Yes, the contacts are in my system contacts and yes, the correct photos in the conversation list are the ones from there. I think I don't have any "signal only" contacts at all if I'm not mistaken 🤷‍♂️

Please let me know if I can give you any more info!

**Comment by CombeeMike** [comment]
Btw, I just realized that notifications of the person shown in the screenshot above now show the correct picture. Not sure if all contacts have been updated in the background by now or if some still show incorrect pictures but at least that one definitely "fixed itself" in the meantime 🤷‍♂️

**Comment by ads3g20** [comment]
> @Cthulku Outside of modding / building your own version of the app (I think there's like, one line you'd comment out in Recipient.java?) no. Showing system contact photos vs profile photos has been discussed in the past, and we've at one point swapped how it works, but ended up changing it back after listening to community feedback. For future reference, the best place to search for stuff like this is over at community.signalusers.org :)

I understand that it's been discussed in the past and I've seen the comments, but the idea of setting a profile picture is redundant given that by default it's overridden by System contact profile pictures. The main reason this is a problem is because for most of us, our contacts have their own google profiles with their Google profile pictures which are synced with the Google Contacts app by default, some of which are very outdated. The only way to overcome this right now seems to be to manually remove/change each contact individually. I believe that showing the ones set on Signal by default would be a good start for all those who are in the process of jumping ship from Whatsapp to Signal, and offering an option to revert to system contact profile pics for all current, hardcore original Signal fans. That's just my opinion but having an option would be an absolute Godsend in this particular case

**Comment by alex-signal** [comment]
@ads3g20 Just a heads up this changed recently. We'll now show profile photos by default with an option to prefer system photos in settings.

**Comment by ads3g20** [comment]
> @ads3g20 Just a heads up this changed recently. We'll now show profile photos by default with an option to prefer system photos in settings. Currently in beta I believe.

Oh my. Thanks for getting back to me so quickly. Very glad to hear that.

Can I also suggest adding very faint divider lines between the separate chats on the opening page of the app? Even though it personally doesn't bother me, the lack of dividers between the chats really confused both my parents and those I've recommended the app to for whatever reason

**Comment by werelord** [comment]
I just had this issue, moving from signal on android 11 to a new phone on android 12; both phones using signal version 5.46.6.  I did the account transfer from the settings.  The images shown on the notifications were of other people in my contacts, not the image associated with the actual contact the message was from.

I uninstalled signal on the new phone, backed up the messages on the old phone (via Chats \ backup) and restored that backup on the new phone; after doing so the images in the notifications were correct..  Not sure what the issue was..

**Comment by g3n35i5** [comment]
I upgraded my phone yesterday from Android 12 (Custom Rom "KOSP") to Android 13 (Custom Rom "ProjectElixir"). Before that I made a data backup in Signal and then restored it on A13. Now I also have the same problem: The profile pictures in the chats belong to other contacts from my address book.

Additionally (maybe related, maybe not, I mention it for completeness) Signal runs extremely slow on the new Rom and crashes sometimes.
